### PR TITLE
ENH list all possible objectives when unmatched pattern option is passed to -p

### LINKED ISSUE #224: ENH: List all possible objectives when unmatched pattern option is passed to -p
this is done for datasets/solvers, but not for objectives

### PR DESCRIPTION
**Pull request by bmalezieux**
All permutations of parameters values are displayed when filters don't match objective attributes. However, I'm not sure of the format of the pattern...

Fixes #224

### CODE REVIEW

File: benchopt/cli/main.py [hunk 1]
```diff
@@ -158,7 +159,7 @@ def run(benchmark, solver_names, forced_solvers, dataset_names,
     solvers_option = ' '.join(['-s ' + s for s in solver_names])
     forced_solvers_option = ' '.join(['-f ' + s for s in forced_solvers])
     datasets_option = ' '.join(['-d ' + d for d in dataset_names])
-    objective_option = ' '.join(['-p ' + p for p in objective_filters])
+    objective_option = ' '.join(['-p ' + f"'{p}'" for p in objective_filters])
```
Comment (tomMoral) [inline benchopt/cli/main.py:162]: Why do you need this?

File: benchopt/benchmark.py [hunk 2]
```diff
@@ -277,6 +278,14 @@ def validate_solver_patterns(self, solver_patterns):
 
         _validate_patterns(all_solvers, solver_patterns, name_type='solver')
 
+    def validate_objective_filters(self, objective_params):
+        "Check that all objective filters match at least one objective setup"
+
+        # List all choices of objective parameters
+        all_params = _list_all_objective_params(self.get_benchmark_objective())
```
Comment (tomMoral) [inline benchopt/benchmark.py:285]: You should simply use `_list_all_parametrized_names`. No need for an extra function here.

Comment (tomMoral) [reply benchopt/benchmark.py:285]: I would call this `all_objectives` to be consistent with the previous functions.

**Review by tomMoral — CHANGES_REQUESTED** [review]
You can reuse `_list_all_parametrize_names` for this PR.

File: benchopt/cli/main.py (hunk 1 above)
Comment (bmalezieux) [reply benchopt/cli/main.py:162]: It seems that this doesn't work for names that contain blank spaces. For instance, `benchopt run -e -p "Convolutional Sparse Coding[reg=0.1]" benchmark_csc/` returns the following error: `Error: Got unexpected extra arguments (Sparse Coding[reg=0.1])` without adding this fix.

File: benchopt/cli/main.py [hunk 3]
```diff
@@ -158,7 +159,7 @@ def run(benchmark, solver_names, forced_solvers, dataset_names,
     solvers_option = ' '.join(['-s ' + s for s in solver_names])
     forced_solvers_option = ' '.join(['-f ' + s for s in forced_solvers])
     datasets_option = ' '.join(['-d ' + d for d in dataset_names])
```
Comment (tomMoral) [inline benchopt/cli/main.py:161]:
```suggestion
    forced_solvers_option = ' '.join([f"-f '{s}'" for s in forced_solvers])
    datasets_option = ' '.join([f"-d '{d}'" for d in dataset_names])
```

File: benchopt/cli/main.py (hunk 1 above)
Comment (tomMoral) [inline benchopt/cli/main.py:162]: Ok, let's do this for all parameters then:
```suggestion
    objective_option = ' '.join([f"-p '{p}'" for p in objective_filters])
```

File: benchopt/benchmark.py [hunk 4]
```diff
@@ -277,6 +277,20 @@ def validate_solver_patterns(self, solver_patterns):
...
+        # List all choices of objective parameters
+        all_objectives = _list_all_parametrized_names(
+            self.get_benchmark_objective()
+        )
+
+        _validate_patterns(
+            all_objectives,
+            objective_params,
+            name_type="objective"
+        )
```
Comment (tomMoral) [inline benchopt/benchmark.py:292]: Doesn't it pass on a single line?

**Review by tomMoral — APPROVED** [review]
A few more comments but overall it looks good!

File: benchopt/benchmark.py (hunk 4 above)
Comment (bmalezieux) [reply benchopt/benchmark.py:292]: No, it is the reason why I changed the name `all_objectives` by something shorter in the previous commit

**Review by tomMoral — APPROVED** [review]
LGTM! thx @bmalezieux

**Pull request merged**